### PR TITLE
Log exceptions in the span

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
Related to https://github.com/smallrye/smallrye-opentracing/pull/23

File: src/main/java/io/opentracing/contrib/interceptors/OpenTracingInterceptor.java [hunk 1]
```diff
@@ -89,6 +90,13 @@ public Object wrap(InvocationContext ctx) throws Exception {
             }
 
             return ctx.proceed();
+        } catch (Throwable t) {
+            logException(scope.span(), t);
+            if (t instanceof Exception) {
+                throw (Exception) t;
+            } else {
+                throw new Exception(t);
```
Comment (pavolloffay) [inline src/main/java/io/opentracing/contrib/interceptors/OpenTracingInterceptor.java]: Why it's wrapped? And why only for errors?

Comment (jpkrohling) [reply src/main/java/io/opentracing/contrib/interceptors/OpenTracingInterceptor.java]: I cannot rethrow it, as `AroundInvoke` methods can only throw `Exception` (Java 6)

Comment (pavolloffay) [reply src/main/java/io/opentracing/contrib/interceptors/OpenTracingInterceptor.java]: Then it's it better to just catch en exception? What happens if the error like OOM is caught? The type of populated exception is changed. it seems rather dangerous to do it.

Comment (jpkrohling) [reply src/main/java/io/opentracing/contrib/interceptors/OpenTracingInterceptor.java]: Agree, I just updated the PR to catch only Exception's

**Comment by jpkrohling** [comment]
@pavolloffay is this OK to be merged?

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**